### ISSUE TITLE
static port is not working

### ISSUE THREAD

**Issue by cliarena** [tristanpemble/nix-nomad#6]
it looks like `to` port works but not the `static` port
it just ignores it and uses a dynamic one
```nix
{ time, ... }:
let
  http = {
    port.http = {
      static = 80;
      to = 8080;
    };
  };
in {
  job.nginx = {
    datacenters = [ "dc1" ];

    group.servers = {
      networks = [ http ];

      task.server = {
        driver = "podman";

        config = {
          image = "nginxinc/nginx-unprivileged:alpine";
          ports = [ "http" ];
        };
        resources = {
          cpu = 10;
          memory = 50;
        };
        services = [{
          name = "nginx-server";
          port = "http";

          tags = [ "pi" "nginx-test-server" ];

          checks = with time; [{
            type = "http";
            path = "/index.html";
            interval = 2 * second;
            timeout = 2 * second;
          }];
        }];
      };
    };
  };
}
```

**Comment by tristanpemble** [comment]
yeah, this is a bug that needs to be fixed. basically, use [`reservedPorts`](https://tristanpemble.github.io/nix-nomad/index.html#opt-job._name_.group._name_.networks._.reservedPorts) for static ports.

I'm gonna leave this open since it's something I do want to fix. at the very least it needs to be documented. it has to do with how HCL <> JSON is not a direct translation and involves some transformations in Nomad's Go code. the Nix schema that is generated tries to approximate it as best it can, but in this case I couldn't find an easy work around.

**Comment by tristanpemble** [comment]
here's the Go tags in the Nomad source repo for me/anyone who looks at this in the future: https://github.com/hashicorp/nomad/blob/1a01e871927284b2c484d7745e1d7df1df6312d8/api/resources.go#L120-L121

and the piece of code that lets you define a static port in the dynamic ports section: https://github.com/hashicorp/nomad/blob/1a01e871927284b2c484d7745e1d7df1df6312d8/jobspec2/parse_job.go#L70-L97

**Comment by GTrunSec** [comment]
I think, we can throw out a warning for this case, then do an alias change between `reservedPorts` and static ports silenced.

**Comment by tristanpemble** [comment]
`port.<label>.static` is now supported. ports with a static value are emitted in the `ReservedPorts` JSON field. ports without a `port.<label>.static` field defined (the default) remain in the output `DynamicPorts`. the existing `reservedPorts.<label>` form remains supported for backward compatibility.
